packages/runtime/test/execute/step.test.ts

```diff
@@ -285,10 +285,10 @@ test.serial('log memory usage', async (t) => {
   t.regex(memory?.message, /\d+mb(.+)\d+mb/i);
 });
 
-test.serial('warn if a non-leaf step does not return state', async (t) => {
+test.serial("warn if a previous step don't return state", async (t) => {
```

Comment:
Hey @doc-han this test isn't quite right :thinking: 

Your edit makes this a single step with two operations in the expression.

But this test is making sure that:
a) when a step in a workflow is finished, check if it returned state
b) if it does not, log a warning

I'd forgotten that we were already doing b), but it's actually broken in this PR!

Here's some log output from production for a workflow where one step doesn't return state:

```
[R/T] ✔ Completed step a in 124ms
[R/T] ❯ Final memory usage: [step 57mb] [system 167mb]
[R/T] ⚠ WARNING: step a did not return a state object. This may cause downstream jobs to fail.
[R/T] ℹ Starting step b
```

Note that these are STEPS, not operations!

What we actually need to do is add an `isNullState` test to `execute/step.ts` line 424, to restore this statement:
```
  if (next.length && !didError && !result) {
    logger.warn(
      `WARNING: step ${stepId} did not return a state object. This may cause downstream jobs to fail.`
    );
  }
```

Then you should be able to revert the changes to this test and it should pass.

Reply:
Currently, null-state doesn't escape the expression execution context. Hence, at the step level, you'll never have it. 
due to:
https://github.com/OpenFn/kit/blob/4376f7b3e615c0b0144deecf6d65d4a03dbb3abf/packages/runtime/src/execute/step.ts#L101-L102

Hence, if the last result before step is null-state. I change that back to undefined so step get's what it expects.